Boxes in the top layer should use the initial containing block as their static-position rectangle

tuankiet65 opened the pull request:
#### b67aa8964b5858168f560d95722d5501800e00d1
<pre>
Boxes in the top layer should use the initial containing block as their static-position rectangle
<a href="https://rdar.apple.com/155495104">rdar://155495104</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=295686">https://bugs.webkit.org/show_bug.cgi?id=295686</a>

Reviewed by Alan Baradlay.

Per resolution [1], boxes in the top layer should use the initial containing block
as their static-position rectangle. Change RenderBlockFlow::adjustOutOfFlowBlock to
set the static block/inline position to 0 if the child is in the top layer.

[1]: <a href="https://github.com/w3c/csswg-drafts/issues/9939">https://github.com/w3c/csswg-drafts/issues/9939</a>

Tests: imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-rtl.html

* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-bottom-left-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-bottom-right-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-top-left-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-top-right-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-rtl.html: Added.
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::adjustOutOfFlowBlock):
    - Rename logicalTop to blockPosition for consistency.
    - Set static block/inline position to 0 if child is in the top layer.
(WebCore::RenderBlockFlow::updateStaticInlinePositionForChild): Deleted.
    - Fold this method into RenderBlockFlow::adjustOutOfFlowBlock, since it needs
      to know if child is in the top layer or not, which is already computed there.
* Source/WebCore/rendering/RenderBlockFlow.h:

Canonical link: <a href="https://commits.webkit.org/312908@main">https://commits.webkit.org/312908@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a7ab35bb434e16424897dde5f17ad3b6282d5666

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/160649 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/34122 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/27223 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/169552 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/115715 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/162518 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/34223 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/34122 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/124583 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/87970 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/163608 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/26832 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/144303 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/105183 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/70b568d5-bf47-4fa6-8cfb-b883aefa917d) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/25884 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/24388 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/17272 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/135578 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/22066 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/172031 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/18908 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/23706 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/132858 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/33796 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/28480 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/132863 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/36071 "Built successfully and passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/33780 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/143861 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/95589 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/27461 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/20676 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/33291 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/100533 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/32790 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/33034 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/32938 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/4ea0c4f9b4ccc584933319b0d383bd7ea9c4c11a)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110873 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20971 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116903 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61141 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112836 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31216 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39122 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84291 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113821 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24933 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99826 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64735 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24289 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | [⏳ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-API-Tests-EWS "Waiting to run tests") | [⏳ 🛠 gtk3-libwebrtc ](https://ews-build.webkit.org/#/builders/GTK-GTK3-LibWebRTC-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94318 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17968 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119705 "Built successfully") | | 
| [⏳ 🛠 🧪 jsc-debug-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-O3-Debug-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-safer-cpp ](https://ews-build.webkit.org/#/builders/Apple-iOS-26-Safer-CPP-Checks-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18030 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93250 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37918 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28173 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93075 "Found 2 new API test failures: /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy, /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38294 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96100 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33892 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38122 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15869 "Found 1 new test failure: imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37811 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43282 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37474 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40810 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39179 "Built successfully") | | | | </details>

tuankiet65 commented:
(changing to draft, we thought this is related to https://bugs.webkit.org/show_bug.cgi?id=295322, but it wasn't)

tuankiet65 committed:
Boxes in the top layer should use the initial containing block as their static-position rectangle
rdar://155495104
https://bugs.webkit.org/show_bug.cgi?id=295686

Reviewed by NOBODY (OOPS!).

Per resolution [1], boxes in the top layer should use the initial containing block
as their static-position rectangle. Change RenderBlockFlow::adjustOutOfFlowBlock to
set the static block/inline position to 0 if the child is in the top layer.

[1]: https://github.com/w3c/csswg-drafts/issues/9939

Tests: imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl.html

* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-left-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-right-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl.html: Added.
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::adjustOutOfFlowBlock):
    - Rename logicalTop to blockPosition for consistency.
    - Set static block/inline position to 0 if child is in the top layer.
(WebCore::RenderBlockFlow::updateStaticInlinePositionForChild): Deleted.
    - Fold this method into RenderBlockFlow::adjustOutOfFlowBlock, since it needs
      to know if child is in the top layer or not, which is already computed there.
* Source/WebCore/rendering/RenderBlockFlow.h:

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/656431d7cf5c9a23ae95cd4d8a81dd48686de20c)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/160002 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/33472 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/26576 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/168880 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/114383 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/206e775a-16b8-4211-8a32-2ca3845ce02b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/161871 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/33575 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/33474 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/124007 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/86979 "1 flakes 2 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/2354a96e-9a3e-47d3-af94-7245d1eeabab) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/162960 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/26262 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/143708 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/104623 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/c9ac7b04-7489-4179-86d4-e1e84a8f5993) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/25315 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/23798 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/16623 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/135006 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/21477 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/171362 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/17370 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/23115 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/132272 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/33148 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/27880 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/132298 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/33133 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/143272 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/91283 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/26918 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/20085 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/32642 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/99039 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/32140 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/32386 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/32290 "Built successfully") | | | | </details>

fantasai reviewed (commented):
I suspect you need some tests for when the ICB and body or popover have different writing modes. I'm not sure which combination is the interesting ones here, though. (I still don't understand how static position coordinates work.) Maybe @alanbaradlay can advise on which cases would be good to cover.

I do suggest throwing in a `writing-mode: sideways-lr` test though, that writing mode has some interesting properties. :)

fantasai reviewed (commented):
I suspect you need some tests for when the ICB and body or popover have different writing modes. I'm not sure which combination is the interesting ones here, though. (I still don't understand how static position coordinates work.) Maybe @alanbaradlay can advise on which cases would be good to cover.

I do suggest throwing in a `writing-mode: sideways-lr` test though, that writing mode has some interesting properties. :)

tuankiet65 committed:
Boxes in the top layer should use the initial containing block as their static-position rectangle
rdar://155495104
https://bugs.webkit.org/show_bug.cgi?id=295686

Reviewed by Alan Baradlay.

Per resolution [1], boxes in the top layer should use the initial containing block
as their static-position rectangle. Change RenderBlockFlow::adjustOutOfFlowBlock to
set the static block/inline position to 0 if the child is in the top layer.

[1]: https://github.com/w3c/csswg-drafts/issues/9939

Tests: imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-rtl.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-ltr.html
       imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-rtl.html

* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-bottom-left-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-bottom-right-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-htb-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-slr-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-srl-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-top-left-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-top-right-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vlr-rtl.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-ltr-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-ltr.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-rtl-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-position/static-position/top-layer-box-uses-icb-vrl-rtl.html: Added.
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::adjustOutOfFlowBlock):
    - Rename logicalTop to blockPosition for consistency.
    - Set static block/inline position to 0 if child is in the top layer.
(WebCore::RenderBlockFlow::updateStaticInlinePositionForChild): Deleted.
    - Fold this method into RenderBlockFlow::adjustOutOfFlowBlock, since it needs
      to know if child is in the top layer or not, which is already computed there.
* Source/WebCore/rendering/RenderBlockFlow.h:

Canonical link: https://commits.webkit.org/312908@main
